projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,540 @@
+          double ac = Math.abs(i - startStopY); // y distance
+          double cb = Math.abs(j - startStopX); // x distance
+          double distance = Math.hypot(ac, cb) * 60;
+          double stopTimeArrival = distance;
+
+          if ((i == startStopX || j == startStopY)) {
+            StopsAround firstStop = stopsAroundList.get(0);
+            ac = Math.abs(i - firstStop.j); // y distance
+            cb = Math.abs(j - firstStop.i); // x distance
+            distance = Math.hypot(ac, cb) * 60;
+            stopTimeArrival = firstStop.arrivalTime;
+          }
+
+          // Find nearest stop
+          for (StopsAround stopAround : stopsAroundList) {
+            ac = Math.abs(i - stopAround.j); // y distance
+            cb = Math.abs(j - stopAround.i); // x distance
+
+            if (distance > Math.hypot(ac, cb)) {
+              distance = Math.hypot(ac, cb) * 60;
+              stopTimeArrival = stopAround.arrivalTime;
+            }
+          }
+
+          heatMap[i][j] = (distance + stopTimeArrival);
+        }
+        if (heatMap[i][j] == 0 && (j != startStopX && i != startStopY)) {
+          System.out.println("Check heatMap assignment");
+          System.out.println("i= " + i + "j=" + j);
+        }
+      }
+    }
+  }
+
+  private boolean inBorders(int x, int y) {
+    // Not in the Sea area
+    // y = -3.244698554·10-3 x2 - 5.987572519·10-1 x + 583.173052
+    if (y
+        > -3.373851571 * (Math.pow(10, -3)) * (Math.pow(x, 2))
+            - 1.457345683 * (Math.pow(10, -1)) * x
+            + 657.4869216) {
+      return true;
+    }
+    return false;
+  }
+
+  public void drawHeatMapOnMap(String imagePath) {
+    // color1 to color2
+    Color color1 = new Color(255, 208, 235);
+    Color color2 = new Color(126, 0, 67);
+
+    // Normalize data
+    double maxData = 0;
+    double minData = heatMap[10][10];
+    for (int i = 0; i < heatMap.length; i++) {
+      for (int j = 0; j < heatMap[0].length; j++) {
+        if (maxData < heatMap[i][j]) {
+          maxData = heatMap[i][j];
+        }
+        if (heatMap[i][j] != 0 && minData > heatMap[i][j]) {
+          minData = heatMap[i][j];
+        }
+      }
+    }
+
+    // Load image to buffer
+    BufferedImage mapImage = null;
+    try {
+      mapImage = (ImageIO.read(new File(imagePath)));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    // Create image
+    final BufferedImage image =
+        new BufferedImage(mapImage.getWidth(), mapImage.getHeight(), BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics2D = image.createGraphics();
+
+    graphics2D.drawImage(mapImage, 0, 0, null);
+
+    // Draw the map
+    for (int i = 0; i < heatMap.length; i++) {
+      for (int j = 0; j < heatMap[0].length; j++) {
+        if (heatMap[i][j] == 0) {
+          int x = getXpixel(startStop.getStopLat(), startStop.getStopLon());
+          int y = getYpixel(startStop.getStopLat(), startStop.getStopLon());
+          int startStopX = x;
```

Comment:
Looks like `startStopX` and `startStopY` variables are never used.

Reply:
Previously i ignore the value of the start point and the distance of the points on the map from this point , and that caused color problems , so now i am calculating to each point the distance from the start point  as the initialize value and then search closer point, if there is one, and change the value.